github/open_close_pr.py

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/python3
+
+import github3
+import argparse
+import logging
+import sys
+import os
+from datetime import datetime
+
+sys.path.insert(0, os.path.join(os.path.dirname(os.path.abspath(__file__)),
+                                '..'))
+from tasks import common
+from prci_github.adapter import GitHubAdapter
+
+
+DEFAULT_BRANCH_NAME = 'nt'
+DEFAULT_FILE_NAME = 'nightly_test.txt'
+DEFAULT_PR_TITLE = '[Nightly Test PR]'
+NEW_BRANCH_REF = 'refs/heads/'
+DEFAULT_COMMIT_MSG = 'File for Nightly Tests'
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+consoleHandler = logging.StreamHandler()
+consoleHandler.setLevel(logging.DEBUG)
+logger.addHandler(consoleHandler)
+
+
+class NightlyTests(object):
+
+    def __init__(self, github_token, repo):
+        github = github3.login(token=github_token)
+        github.session.mount('https://api.github.com',
+                             GitHubAdapter())
+
+        self.__repo = github.repository(repo['owner'],
+                                        repo['name'])
+
+    def __create_branch(self):
+        last_commit = next(self.__repo.commits())
+        branch_name = '{}_{}'.format(DEFAULT_BRANCH_NAME,
+                                     datetime.now().strftime("%Y%m%d_%H%M%S"))
+
+        new_branch_ref = '{}{}'.format(NEW_BRANCH_REF, branch_name)
+        self.__repo.create_ref(new_branch_ref, last_commit.sha)
+
+        return branch_name
+
+    def open_pr(self, args):
+        # a branch is created and a commit is done in order to use them
+        # to create a new PR. The content of the commit doesn't matter
+        branch_name = self.__create_branch()
+        self.__repo.create_file(DEFAULT_FILE_NAME, DEFAULT_COMMIT_MSG,
+                                branch_name.encode(), branch_name)
+
+        pr_title = '{} {}'.format(DEFAULT_PR_TITLE, branch_name)
+
+        logger.info("A new PR against %s/%s will be created with "
+                    "the title %s", self.__repo.owner.login,
+                    self.__repo.source.name, pr_title)
+
+        head = '{}:{}'.format(self.__repo.owner.login, branch_name)
+        pr = self.__repo.create_pull(pr_title, 'master', head)
+        logger.info("PR %s created", pr.number)
+
+    def close_pr(self, args):
```

Comment:
How do you propose we automate closing the PR when you need to specify the PR number? Would it be possible to detect it from branch name or title instead?

Reply:
How I thought we could solve it: we'll need another tool that checks the status of opened PRs and produces the difference between the result of last Nightly Test PR and the current one. 

Probably, this tool will have to go over the opened PRs and "discover" which one is the "Nightly Tests PR", right? Then, it'll compare with the last result, add a comment with it and call the `close_pr` using the PR number.

What do you think?